### PR TITLE
Create `conf-*` packages for OpenGL related system libraries

### PR DESCRIPTION
**Pull request by Leonidas-from-XIV**
Given the depexts for OpenGL can be kinda hairy with complicated dependency names, this PR attempts to move the dependencies of `lablgl` into its own `conf-*` packages (and updates raylib to use them where there is agreement).

The libraries in question:

* libGL
* libGLU (part of Mesa but still a separate library)
* freeglut (implementation of GLUT)

### CODE REVIEW

**Review by jmid — COMMENTED** [review]
This is nice! :smiley: 

A small plea:
I recently got tired of conf-* packages failing for me on Mint (an Ubuntu derivative distro) and hence started trawling the 240-something conf-* packages to fix them. I'm only at no.145 or so though :sweat_smile: 

Can I convince you to add `os-family="ubuntu"` cases and conditions, that just mirror the `"debian"` ones? :pray: 
I just checked: freeglut3-dev, mesa-common-dev, and libglu1-mesa-dev are all available (originating from Ubuntu :shrug:)

File: packages/conf-libgl/conf-libgl.1/opam [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  ["pkg-config" "--exists" "gl"]
+]
+depends: [
+  "conf-pkg-config" {build}
+]
+depexts: [
+  ["mesa-common-dev"] {os-family = "debian"}
+  ["mesa-libGL-devel"] {os-distribution = "fedora"}
+  ["mesa-dev"] {os-distribution = "alpine"}
+  ["libgl"] {os-family = "arch"}
```
Comment (jmid) [inline packages/conf-libgl/conf-libgl.1/opam:16]:
```
+   ["libglvnd" {os = "freebsd"}
```
according to https://www.freshports.org/graphics/libglvnd
(prev. libgl was apparently part of https://www.freshports.org/graphics/mesa-libs
and before that https://www.freshports.org/graphics/libGL)

File: packages/conf-libglu/conf-libglu.1/opam [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  ["pkg-config" "--exists" "glu"]
+]
+depends: [
+  "conf-pkg-config" {build}
+]
+depexts: [
+  ["libglu1-mesa-dev"] {os-family = "debian"}
+  ["mesa-libGLU-devel"] {os-distribution = "fedora"}
+  ["glu"] {os-family = "arch"}
+  ["glu-dev"] {os-distribution = "alpine"}
```
Comment (jmid) [inline packages/conf-libglu/conf-libglu.1/opam:17]:
```
+    ["libGLU"] {os = "freebsd"}
```
according to https://www.freshports.org/graphics/libGLU

Comment (Leonidas-from-XIV) [reply packages/conf-libglu/conf-libglu.1/opam:18]: Done!

File: packages/conf-libgl/conf-libgl.1/opam (hunk 1 above)
Comment (Leonidas-from-XIV) [reply packages/conf-libgl/conf-libgl.1/opam:17]: Yeah, this one is a bit weird, in Arch it is also libglvnd which seems mostly like a library that redirects to your actual GL implementation, be it Mesa, NVIDIA or something else. I'll try and see what the CI says!

**Comment by Leonidas-from-XIV** [comment]
> Can I convince you to add os-family="ubuntu" cases and conditions, that just mirror the "debian" ones?

Sure! To be honest I sort of expected Ubuntu and Mint to be `os-family` Debian, given it takes (most) packages from Debian.

It's all a bit confusing wrt families. Is CentOS/Rocky/Alma the RHEL family? Is Fedora part of it? Is Oraclelinux part of it? I wish there was an overview of what is what.

File: packages/conf-libgl/conf-libgl.1/opam (hunk 1 above)
Comment (Leonidas-from-XIV) [reply packages/conf-libgl/conf-libgl.1/opam:17]: CI is happy with that; resolving :)

**Comment by Leonidas-from-XIV** [comment]
I've added exceptions for distributions that don't have `.pc` files to not use pkg-config, similar to `conf-gles2`. Debian 12 has them, Debian <12 is missing some but it is a bit tedious to add `os-version` checks for each of the failures accurately. Given it is just an installation check; I don't think it is that vital.

This seems to have worked on Debian and Ubuntu but somehow `oraclelinux` still runs the code? I don't quite know what is going on; `os-distribution = "oraclelinux"` seems to be the name that is used in opam-repository generally.

**Comment by jmid** [comment]
> This seems to have worked on Debian and Ubuntu but somehow `oraclelinux` still runs the code? I don't quite know what is going on; `os-distribution = "oraclelinux"` seems to be the name that is used in opam-repository generally.

A quick skim at the conf-packages I recently fiddled with (and that actually make an attempt at oracle support), reveals that the following all use `{os-distribution = "ol"}` and achieve green CI lights on it:
- conf-openssl
- conf-ruby
- conf-libtool
- conf-jq

(just my 0.05$, not accounting for inflation)

**Comment by Leonidas-from-XIV** [comment]
@jmid Thanks, that seemed to be the solution to the mystery! :) I picked up `oraclelinux` from other conf packages but seems like the right value is indeed `ol`.

To avoid this confusion in the future I opened #25511 to replace all occurences of `oraclelinux` with `ol`.

**Comment by Leonidas-from-XIV** [comment]
Turns out there is a wiki page with documentation on the values: https://github.com/ocaml/opam-repository/wiki/Depexts-os-distribution---os-family-values

**Comment by mseri** [comment]
Thanks!

**Pull request merged**